pymatviz/ptable.py

```diff
@@ -703,34 +701,39 @@ def histogram(
 
 
 def ptable_heatmap(
-    values: ElemValues,
+    data,
     *,
-    log: bool | Normalize = False,
+    # TODO: to be removed
     ax: plt.Axes | None = None,
-    count_mode: ElemCountMode = ElemCountMode.composition,
-    cbar_title: str = "Element Count",
-    cbar_range: tuple[float | None, float | None] | None = None,
-    cbar_coords: tuple[float, float, float, float] = (0.18, 0.8, 0.42, 0.05),
-    cbar_kwargs: dict[str, Any] | None = None,
-    colorscale: str = "viridis",
-    show_scale: bool = True,
-    show_values: bool = True,
+    exclude_elements: Sequence[str] = (),  # control by elements in data
+    count_mode: ElemCountMode = ElemCountMode.composition,  # separate data preprocess stage
+    # Heatmap specific
+    heat_mode: Literal["value", "fraction", "percent"] | None = "value",
+    tile_size: float | tuple[float, float] = 0.9,  # control by child_kwargs
     infty_color: str = "lightskyblue",
     na_color: str = "white",
-    heat_mode: Literal["value", "fraction", "percent"] | None = "value",
-    fmt: str | Callable[..., str] | None = None,
-    cbar_fmt: str | Callable[..., str] | None = None,
-    text_color: str | tuple[str, str] = "auto",
-    exclude_elements: Sequence[str] = (),
     zero_color: str = "#eff",  # light gray
     zero_symbol: str | float = "-",
-    text_style: dict[str, Any] | None = None,
-    label_font_size: int = 16,
-    value_font_size: int = 12,
-    tile_size: float | tuple[float, float] = 0.9,
+    show_values: bool = True,
+    log: bool | Normalize = False,
+    # Figure-scope
     f_block_voffset: float = 0.5,
     hide_f_block: bool | None = None,
```

Comment:
I kind of want to change `hide_f_block` to something like `bool | Literal["AUTO"] = "AUTO"`, and detect if data if present for f-block by default. I think this is more self-descriptive than `None`?

Reply:
yes, that's more descriptive 👍